java-sdk/src/test/java/com/netflix/conductor/sdk/workflow/executor/task/AnnotatedWorkerTests.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.sdk.workflow.executor.task;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.sdk.workflow.task.InputParam;
+import com.netflix.conductor.sdk.workflow.task.OutputParam;
+import com.netflix.conductor.sdk.workflow.task.WorkerTask;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AnnotatedWorkerTests {
+
+    static class Car {
+        String brand;
+
+        String getBrand() {
+            return brand;
+        }
+
+        void setBrand(String brand) {
+            this.brand = brand;
+        }
+    }
+
+    static class Bike {
+        String brand;
+
+        String getBrand() {
+            return brand;
+        }
+
+        void setBrand(String brand) {
+            this.brand = brand;
+        }
+    }
+
+    static class CarWorker {
+        @WorkerTask("test_1")
+        public @OutputParam("result") List<Car> doWork(@InputParam("input") List<Car> input) {
+            return input;
+        }
+    }
+
+    @Test
+    @DisplayName("it should handle null values when InputParam is a List")
```

Comment:
With this test I was able to reproduce the issue before the fix.

When `inputData` was `null` (in the `Task`) this caused a NPE in the annotatedWorker before reaching its execution method.

![image](https://user-images.githubusercontent.com/4755315/204310605-95e0e8d4-53da-4e78-9211-f540a5c728cf.png)